holopy/scattering/theory/mielensfunctions.py

```diff
@@ -264,11 +264,60 @@ def _interpolate_and_eval_mielens_i_n(self, krho, n=0):
             window_breakpoints=window_breakpoints)
         return interpolator(krho)
 
+    def _calculate_phase(self):
+        return self.particle_kz * (1 - self._quad_pts)
+
     def _check_parameters(self):
-        must_be_specified = ['particle_kz', 'index_ratio',
-                             'size_parameter', 'lens_angle']
-        if any([getattr(self, p) is None for p in must_be_specified]):
-            raise ValueError("{} must be specified.".format(must_be_specified))
+        if any([getattr(self, p) is None for p in self.must_be_specified]):
+            msg = "{} must be specified.".format(self.must_be_specified)
+            raise ValueError(msg)
```

Comment:
HoloPy actually has a `MissingParameter` exception class you might want to use here.

Reply:
Fixed by 08b8338d